Userland/Libraries/LibSQL/Database.cpp

```diff
@@ -18,31 +18,53 @@
 namespace SQL {
 
 Database::Database(String name)
-    : m_heap(Heap::construct(name))
+    : m_heap(Heap::construct(move(name)))
     , m_serializer(m_heap)
-    , m_schemas(BTree::construct(m_serializer, SchemaDef::index_def()->to_tuple_descriptor(), m_heap->schemas_root()))
-    , m_tables(BTree::construct(m_serializer, TableDef::index_def()->to_tuple_descriptor(), m_heap->tables_root()))
-    , m_table_columns(BTree::construct(m_serializer, ColumnDef::index_def()->to_tuple_descriptor(), m_heap->table_columns_root()))
 {
+}
+
+ErrorOr<void> Database::open()
+{
+    if (auto error_maybe = m_heap->open(); error_maybe.is_error())
+        return error_maybe.error();
```

Comment:
I think this can be written as:
```suggestion
    TRY(m_heap->open());
```